lib/extendedkey.ex

```diff
@@ -303,28 +411,39 @@ defmodule Bitcoinex.ExtendedKey do
   end
 
   @doc """
-    serialize_extended_key takes an extended key
+    serialize takes an extended key
     and returns the binary
   """
-  @spec serialize_extended_key(t()) :: binary
-  def serialize_extended_key(xkey) do
+  @spec serialize(t()) :: binary
+  def serialize(xkey) do
     (xkey.prefix <>
        xkey.depth <> xkey.parent_fingerprint <> xkey.child_num <> xkey.chaincode <> xkey.key)
     |> Base58.append_checksum()
   end
 
+  @doc """
+    serialize takes an extended key
+    and returns the binary without the checksum appended
+    (used for PSBT encoding)
+  """
+  @spec serialize(t(), atom) :: binary
+  def serialize(xkey = %__MODULE__{}, :no_checksum) do
```

Comment:
it will be great to merge &serialize/1  and &serialize/2 into one `&serialize/2` with type signature like
```
@spec serialize(t(), list({:with_checksum?,  boolean}) :: binary
  def serialize(xkey, opts \\ []) do
     with_checksum? =  Keyword.get(opts, :with_checksum?, true)
      extended_key_without_checksum_bin = (xkey.prefix <>
       xkey.depth <> xkey.parent_fingerprint <> xkey.child_num <> xkey.chaincode <> xkey.key)
    case with_checksum? do
       true ->
          Base58.append_checksum(extended_key_without_checksum_bin)
        false ->
           extended_key_without_checksum_bin
    end
    |> 
  end
```
I think it's more elixir-idiomatic, better typing for user using dialyzer and less code duplication